src/tmpfiles/tmpfiles.c

```diff
@@ -2632,6 +2742,64 @@ static int find_gid(const char *group, gid_t *ret_gid, Hashmap **cache) {
         return name_to_gid_offline(arg_root, group, ret_gid, cache);
 }
 
+static int parse_age_by_from_arg(const char *age_by_str, Item *item) {
+        const char *s = NULL;
+        AgeBy ab_f = 0, ab_d = 0;
+
+        static const struct {
+                char age_by_chr;
+                AgeBy age_by_flag;
+        } age_by_types[] = {
+                { 'a', AGE_BY_ATIME },
+                { 'b', AGE_BY_BTIME },
+                { 'c', AGE_BY_CTIME },
+                { 'm', AGE_BY_MTIME },
+        };
+
+        assert(age_by_str);
+        assert(item);
+
+        s = age_by_str;
+
+        if (isempty(s))
+                return -EINVAL;
+
+        do {
+                /* Ignore whitespace. */
+                if (IN_SET(*s, ' ', '\t'))
+                        continue;
+
+                size_t i;
+                for (i = 0; i < ELEMENTSOF(age_by_types); i++) {
```

Comment:
maybe write as:

```c
for (size_t i = 0; …
```

Reply:
So, `i` is used below to check for an invalid character.

Reply:
then maybe move the declaration to the beginning of the block, i.e. right after the `for (…) {`

otherwise this is a bit confusing, given that you use it at various places but only define it close to one of them.